infrastructure/cdn-in-a-box/traffic_ops/run-perl.sh

```diff
@@ -0,0 +1,149 @@
+DBCONF
+
+cat >$TO/conf/production/database.conf <<-DATABASECONF
+{
+	"description": "PostgreSQL database on $DB_SERVER port $TO_PORT",
+	"dbname": "$DB_NAME",
+	"hostname": "$DB_SERVER",
+	"user": "$DB_USER",
+	"password": "$DB_PASSWORD",
+	"port": "$DB_PORT",
+	"ssl": false,
+	"type": "Pg"
+}
+DATABASECONF
+
+crt=$TO/conf/trafficops.crt
+key=$TO/conf/trafficops.key
+openssl req -newkey rsa:2048 -nodes -keyout $key -x509 -days 365 -out $crt -subj "/C=$CERT_COUNTRY/ST=$CERT_STATE/L=$CERT_CITY/O=$CERT_COMPANY"
+
+cat >$TO/conf/cdn.conf <<-CDNCONF
+{
+    "hypnotoad" : {
+        "listen" : [
+            "https://trafficops-perl:60443?cert=$crt&key=$key&verify=0x00&ciphers=AES128-GCM-SHA256:HIGH:!RC4:!MD5:!aNULL:!EDH:!ED"
+        ],
+        "user" : "trafops",
+        "group" : "trafops",
+        "heartbeat_timeout" : 20,
+        "pid_file" : "/var/run/traffic_ops.pid",
+        "workers" : 12
+    },
+    "traffic_ops_golang" : {
+        "port" : "443",
+        "proxy_timeout" : 60,
+        "proxy_keep_alive" : 60,
+        "proxy_tls_timeout" : 60,
+        "proxy_read_header_timeout" : 60,
+        "read_timeout" : 60,
+        "read_header_timeout" : 60,
+        "write_timeout" : 60,
+        "idle_timeout" : 60,
+        "log_location_error": "/var/log/traffic_ops/error.log",
+        "log_location_warning": "",
+        "log_location_info": "",
+        "log_location_debug": "",
+        "log_location_event": "/var/log/traffic_ops/access.log",
+        "max_db_connections": 20,
+        "backend_max_connections": {
+            "mojolicious": 4
+        }
+    },
+    "cors" : {
+        "access_control_allow_origin" : "*"
+    },
+    "to" : {
+        "base_url" : "https://$TO_HOST",
+        "email_from" : "no-reply@traffic-ops-domain.com",
+        "no_account_found_msg" : "A Traffic Ops user account is required for access. Please contact your Traffic Ops user administrator."
+    },
+    "portal" : {
+        "base_url" : "http://localhost:8080/!#/",
+        "email_from" : "no-reply@traffic-portal-domain.com",
+        "pass_reset_path" : "user",
+        "user_register_path" : "user"
+    },
+    "secrets" : [
+        "$TO_SECRET"
+    ],
+    "geniso" : {
+        "iso_root_path" : "$TO/public"
+    },
+    "inactivity_timeout" : 60
+}
+CDNCONF
+
+# ensure db is available for DB_USER
+export PGPASSWORD="$DB_PASSWORD"
+until psql -h "$DB_SERVER" -U "$DB_USER" -p "$DB_PORT" -c '\l'; do
+  >&2 echo "Postgres is unavailable - sleeping"
+  sleep 1
+done
+
+sleep 3 # give it a few extra seconds
+
+psql -h "$DB_SERVER" -U "$DB_USER" -p "$DB_PORT" -c '\du'
+
+>&2 echo "Postgres is up"
+
+
+export PERL5LIB=$TO/lib:$TO/local/lib/perl5
+cd $TO || (echo "NO $TO found" && exit 1)
+
+chown -R trafops:trafops .
+
+# needed for goose
+export GOPATH=/opt/traffic_ops/go
+export PATH=$PATH:$GOPATH/bin:/usr/local/go/bin
+
+./db/admin.pl -env production upgrade
```

Comment:
Do we need to run postinstall or todb_bootstrap before this?

Reply:
too many components involved in postinstall,   which is why I decided not to use it.  but yes, todb_bootstrap would help.